webui/react/src/services/utils.ts

```diff
@@ -121,31 +110,62 @@ export const consumeStream = async <T = unknown>(
     if (process.env.IS_DEV) options.credentials = 'include';
 
     const response = await fetch(serverAddress(fetchArgs.url), options);
-    const reader = ndjsonStream(response.body).getReader();
+    if (!response.body) throw new DetError(`Unable to fetch stream from ${fetchArgs.url}.`);
+
+    const decoder = new TextDecoder();
+    const reader = response.body.getReader();
+    let buffer = '';
+    let isCancelled = false;
 
     // Cancel reader if an abort signal is received.
     if (options?.signal) {
       const signal: AbortSignal = options.signal;
       const abortHandler = () => {
-        reader.cancel();
         signal.removeEventListener('abort', abortHandler);
+        isCancelled = true;
       };
       signal.addEventListener('abort', abortHandler);
     }
 
-    let result;
-    while (!result || !result.done) {
-      result = await reader.read();
-      if (result.done) return;
-      if (result.value.error) {
-        throw result.value.error;
-      } else {
-        onEvent(result.value.result);
+    const handleStreamError = (e: unknown) => handleError(e, { silent: true });
+    const handleStreamLine = (line: string) => {
+      if (isCancelled) return;
+      try {
+        const ndjson = JSON.parse(line);
+        onEvent(ndjson.result);
+      } catch {
+        // JSON parsing occurred, no-op.
       }
-    }
+    };
+    const handleStreamRead = (result: ReadableStreamDefaultReadResult<ArrayBuffer>): unknown => {
+      if (result.done || isCancelled) {
+        // Process any data buffer remainder.
+        buffer = buffer.trim();
+        if (buffer.length !== 0) handleStreamLine(buffer);
```

Comment:
![image](https://user-images.githubusercontent.com/28383080/165927406-8d7dcd3b-de41-49fc-83c7-ad96df0ad875.png)

With this I never actually see 'cancelled' in the console. Don't know if maybe there's a closure thing here.

Looks like I can get state update warnings pretty easily on both this PR and on master when I try, so maybe not relevant, but want to bring up just in case.

https://user-images.githubusercontent.com/28383080/165927747-83c940fe-a9db-41e6-8c95-2136f55db4d8.MOV